core/controllers/admin_test.py

```diff
@@ -1122,6 +1122,39 @@ def test_send_dummy_mail(self):
                 generated_response['error'], 'This app cannot send emails.')
 
 
+class UpdateUsernameHandlerTest(test_utils.GenericTestBase):
+    """Tests for updating usernames."""
+
+    def setUp(self):
+        super(UpdateUsernameHandlerTest, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+
+    def test_update_username(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+        csrf_token = self.get_new_csrf_token()
+
+        current_username, new_username = self.ADMIN_USERNAME, 'newUsername'
+        user_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+
+        self.put_json(
+            '/updateUsername',
```

Comment:
We typically don't use case-sensitive urls. I think the other example you are following in main.py is wrong; please could you fix that too?

Follow the existing URL pattern otherwise.

Reply:
Done.